### ISSUE TITLE
Clean up workflow - branching, merging etc

### ISSUE THREAD

**Issue by partikus** [php-refactor/mail-sender#12]
@php-refactor 
Hi all, I'd like to propose to change the workflow. 
IMO we should create ``develop`` branch, and after each session we will push changes there, not into ``master ``. 
Then doing homework will be based on the ``develop`` branch. And only proper changes should be merge into master. But feel free to propose others workflows.

**Comment by dominikhajduk** [comment]
What when after meeting we produce code which is not working? Should also push to develop and fix with new PR?

**Comment by partikus** [comment]
@dominikhajduk 
Yes, as name suggests there is code which does not work, and as a homework someone can fix that and create new PR to master

**Comment by dominikhajduk** [comment]
@partikus for me sound good. Also my proposal is to merge after at lease two comments that code looks ok. So we're waiting for second one (my is first) and will create develop branch.

**Comment by dominikhajduk** [comment]
@php-refactor Question about flow. For me ambiguous is how we should now use `develop` and `master`.
Master is generally clear, we're putting there main working code. But what about develop? After meeting should we put code into develop and use as code base for pull request to merge? (thats if correct is clear for me). Then after merge mentioned code into master clear develop before next meeting? Am I right?

**Comment by PorkOne** [comment]
:+1:

**Comment by partikus** [comment]
@dominikhajduk @PorkOne @php-refactor  We should consider whether we can override changes from 'develop' branch after each session? Because I'm afraid we won't able to stay with on simple flow.

**Comment by PorkOne** [comment]
What do you mean exactly?

**Comment by partikus** [comment]
:+1:  I can agree with Dominik that after each session we should push changes into ``develop`` branch, even it's unstable, and at home everyone can improve that and create PRs.

**Comment by dominikhajduk** [comment]
@php-refactor @partikus or create meeting branch and remove them after approve pull request. The key is to allow contributors work on code produced during meeting but not merge them into master (after meeting code can be not ready for master).
